### PR TITLE
[nodejs] create new weblog for express 5

### PR DESCRIPTION
**Pull request by IlyasShabi**
## Motivation

Express has released a new version, v5, which includes some breaking changes compared to v4. We have a [PR ready](https://github.com/DataDog/dd-trace-js/pull/4913) for review and merging on `dd-trace-js`. Additionally, we need to create a weblog for Express 5 to identify potential breaking changes and ensure our instrumentation remains compatible.

## Changes

- Change express4 folder name to express
- Change all express dockerfile configuration
- Exclude graphql and mongo sanitize from express 5

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

File: utils/scripts/load-binary.sh
Comment (IlyasShabi) [inline utils/scripts/load-binary.sh:218]: Should be removed after review

File: utils/build/docker/nodejs/express4-typescript.Dockerfile
Comment (CarlesDD) [inline utils/build/docker/nodejs/express4-typescript.Dockerfile:33]: `express4-typescript` application has these dependencies already declared in its [package.json](https://github.com/DataDog/system-tests/blob/main/utils/build/docker/nodejs/express4-typescript/package.json#L19-L22). Why do we need to explicitly install them when building the image?

File: manifests/nodejs.yml
Comment (uurien) [reply manifests/nodejs.yml:566]: As express5 is not yet supported, I think that you can't set any version <5.29.0 or <4.53.0 (next release) for any test.

File: utils/build/docker/nodejs/express4.Dockerfile
Comment (uurien) [inline utils/build/docker/nodejs/express4.Dockerfile:17]: as package-lock.json will not work for these dependencies, should we fix the versions?

```suggestion
RUN npm install "express@4.17.2" "apollo-server-express@3.13.0" "express-mongo-sanitize@2.2.0"
```

File: utils/build/docker/nodejs/express5.Dockerfile
Comment (uurien) [inline utils/build/docker/nodejs/express5.Dockerfile:17]: as package-lock.json will not work for these dependencies, should we fix the versions?

```
RUN npm install "express@5.0.1"
```

**Review by CarlesDD — COMMENTED** [review]
`missing_feature` decorator for [this test](https://github.com/DataDog/system-tests/blob/main/tests/test_data_integrity.py#L82) should be addressed to check the new weblog variant for express5.

File: utils/build/docker/nodejs/express/package.json
Comment (CarlesDD) [inline utils/build/docker/nodejs/express/package.json:1]: Nit: Should we change the application name from `express4` to `express`?

File: utils/build/docker/nodejs/express4-typescript.Dockerfile
Comment (IlyasShabi) [reply utils/build/docker/nodejs/express4-typescript.Dockerfile:33]: Good catch, should not do this for express4-typescript

**Review by cbeauchesne — COMMENTED** [review]
I'm not a node expert, but would it be more reliable to have a package/lock per weblog, rather a common one + a dedicated `npm install` ?

File: utils/build/docker/nodejs/express/package.json
Comment (IlyasShabi) [reply utils/build/docker/nodejs/express/package.json:1]: Done

File: utils/build/docker/nodejs/express4-typescript.Dockerfile
Comment (IlyasShabi) [reply utils/build/docker/nodejs/express4-typescript.Dockerfile]: Done

File: utils/build/docker/nodejs/express4.Dockerfile
Comment (IlyasShabi) [reply utils/build/docker/nodejs/express4.Dockerfile]: Done

File: utils/build/docker/nodejs/express5.Dockerfile
Comment (IlyasShabi) [reply utils/build/docker/nodejs/express5.Dockerfile]: Done

File: tests/appsec/iast/source/test_parameter_name.py
Comment (cbeauchesne) [inline tests/appsec/iast/source/test_parameter_name.py:43]: This change is surprising, it goes from `weblog_variant == "express4"` to `weblog_variant not in ["express4" ...)`. Is it intended ?

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
Some CI failures to fix, probably related to my comment.

File: tests/appsec/iast/source/test_parameter_name.py
Comment (IlyasShabi) [reply tests/appsec/iast/source/test_parameter_name.py:43]: Thanks I will fix this, the feature is currently marked as missing in `nodejs.yml`
The failing test in nodejs prod is expected as my PR hasn't been merged yet.

Comment (cbeauchesne) [reply tests/appsec/iast/source/test_parameter_name.py:43]: ok, passing the PR as draft to unstress my noti stack. You can set it back to normal once it's ready to be merged.

File: manifests/nodejs.yml
Comment (simon-id) [reply manifests/nodejs.yml]: you could probably do a yaml anchor like `*express5_release_5_30_0`

Comment (IlyasShabi) [reply manifests/nodejs.yml]: I used the one for next release

File: utils/build/docker/nodejs/express/package-lock.json
Comment (IlyasShabi) [reply utils/build/docker/nodejs/express/package-lock.json:1]: yes why not

File: utils/build/docker/nodejs/express4.Dockerfile
Comment (simon-id) [inline utils/build/docker/nodejs/express4.Dockerfile:17]: should we not install latest express ? like "express@4" ?

File: utils/build/docker/nodejs/express5.Dockerfile
Comment (simon-id) [inline utils/build/docker/nodejs/express5.Dockerfile:17]: same thing here, why not just express@5 ?

File: manifests/nodejs.yml
Comment (simon-id) [inline manifests/nodejs.yml:1]: I'm really unsure what version we should put for express5 on all tests

File: utils/build/docker/nodejs/express4.Dockerfile
Comment (simon-id) [inline utils/build/docker/nodejs/express4.Dockerfile:39]: hey, why remove that little reminder on how to use docker lol, someone put it there probably because it was useful to them no ?

File: utils/build/docker/nodejs_otel/express4-otel.Dockerfile
Comment (simon-id) [inline utils/build/docker/nodejs_otel/express4-otel.Dockerfile:19]: you're using the caret ^ in this dockerfile but not the other one, is that intended ?

File: utils/build/docker/nodejs/express4.Dockerfile
Comment (IlyasShabi) [reply utils/build/docker/nodejs/express4.Dockerfile:39]: I did remove it because there is no `nodejs.datadog.Dockerfile` but If you think that it's needed as a little reminder I can put it back

File: manifests/nodejs.yml
Comment (IlyasShabi) [reply manifests/nodejs.yml:1]: Same for me, I added `express5` wherever I found that we specify the `express4` version for a feature.

However, for something like this:
```
Test_PathParams:
   '*': v2.0.0
```
Should we keep it as is, or should we specify `express5` versions explicitly?

@cbeauchesne Do you have any thoughts on this?

Comment (simon-id) [reply manifests/nodejs.yml:1]: To give you some context, express 5.0.0 was released recently, and we're releasing support for it soon. IMO we should have all tests that are not framework independent, be set to the latest dd-trace for express5, because otherwise when running tests with older versions we'll end up with a lot of failures.

@IlyasShabi maybe a way to do this would be to try to run this PR on master, list all the tests that are failing, and add `express5: xxx` to them

Comment (IlyasShabi) [reply manifests/nodejs.yml:1]: - `express4` is 2.0.0
- `express5` is not released yet, but `Test_PathParams` might be related to our changes to support express5 in dd-trace or it might not be.

Comment (IlyasShabi) [reply manifests/nodejs.yml:1]: @simon-id No need to run it on master, failing tests can be found on (nodejs, prod) runner

Comment (simon-id) [reply manifests/nodejs.yml:1]: Ok I've double checked every tests, and i found you skipped some that were actually passing before. I found some other inconsistencies that i'm cleaning up rn

File: manifests/nodejs.yml
Comment (simon-id) [inline manifests/nodejs.yml:835]: APM should choose what to do here

File: manifests/nodejs.yml
Comment (simon-id) [inline manifests/nodejs.yml:618]: APM should decide what to do with all the integrations/ tests

File: manifests/nodejs.yml
Comment (simon-id) [inline manifests/nodejs.yml:121]: this test should pass

Comment (simon-id) [inline manifests/nodejs.yml:121]: this test should pass

File: manifests/nodejs.yml
Comment (simon-id) [inline manifests/nodejs.yml:78]: this should passes in earlier versions

File: manifests/nodejs.yml
Comment (simon-id) [reply manifests/nodejs.yml:121]:
```suggestion
        test_header_injection.py:
          TestHeaderInjection:
            '*': *ref_4_21_0
```

File: manifests/nodejs.yml
Comment (simon-id) [reply manifests/nodejs.yml:78]:
```suggestion
        Test_Schema_Request_FormUrlEncoded_Body:
          '*': *ref_4_21_0
```

File: manifests/nodejs.yml
Comment (simon-id) [reply manifests/nodejs.yml:394]: keep old version

**Comment by simon-id** [comment]
Reminder to remove the line that makes CI run on the feature branch before merging

**Review by wconti27 — APPROVED** [review]
LGTM

File: utils/scripts/load-binary.sh
Comment (simon-id) [reply utils/scripts/load-binary.sh:218]:
```suggestion
    echo "DataDog/dd-trace-js#master" > nodejs-load-from-npm
```

**Review by cbeauchesne — APPROVED** [review]

**Pull request merged**
